### PR TITLE
Make default textarea grow based on text content unless there is an explicit height

### LINKED ISSUE #2475: TextArea's input is smaller than the component's height property
# 🐛 Bug Report 

The `TextArea` component no longer uses the `height` property correctly. The input is smaller than component's size.

## 🤔 Expected Behavior

`TextArea` should have its input display in the proper height.

## 😯 Current Behavior

In React Spectrum v3.13.0 using `height="size-3000"` provided an input that is larger and matched the component's size. The new version (> 3.14.0) has the correct component size, but the input is smaller.

image

## 🔦 Context

We are using the component in a Dialog, With the component and input sizes being mismatched it creates too much whitespace within the Dialog.

## 💻 Code Sample

I copied the `TextArea` component directly from our [codebase ](https://git.corp.adobe.com/CloudEng/edgecert-ui/blob/main/src/react/home/entry-point-details/history/AddNote.jsx#L25) into a [codesandbox.io sandbox](https://codesandbox.io/s/dazzling-germain-713qj?file=/src/App.js) and you can observe the undesired behavior when switching between React Spectrum v3.13.0 and v3.14.0/1

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-spectrum   | 3.14.1
| Browser          |  Chrome 94.0.4606.81
| Operating System | Windows 10 Pro

## 🧢 Your Company/Team

Adobe/Cloud Enablement Services - Management Tooling

### LINKED ISSUE #3142: Update TextArea resize behavior
# 🙋 Feature Request

As per discussion: https://github.com/adobe/react-spectrum/pull/2515#issuecomment-1130341691, the team has decided to remove manual resizing from the TextArea and have the TextArea grow to fit its contents if a static height isn't provided.
https://spectrum.adobe.com/page/text-area/#Height

Making these changes should make fixing https://github.com/adobe/react-spectrum/issues/2475 easier since we won't need to handle so many manual resizing cases

## 🤔 Expected Behavior

TextArea should not have a resize handle and should grow to fit its contents if a static height isn't provided.

## 🧢 Your Company/Team

RSP

### PR DESCRIPTION
**Pull request by devongovett**
Closes #2515. Closes #2475. Closes #3142.

This removes the draggable resizer and instead makes the default textarea grow based on the text content similar to isQuiet. This is more in line with the [Spectrum guidelines](https://spectrum.adobe.com/page/text-area/#Height).

It also fixes the issue where if an explicit height is set it was not propagated correctly. For isQuiet, this is treated more like a max height since the line is meant to be directly under the text.

### CODE REVIEW

File: packages/@react-spectrum/label/src/Field.tsx [hunk 1]
```diff
@@ -79,12 +79,23 @@ function Field(props: SpectrumFieldProps, ref: RefObject<HTMLElement>) {
...
     );
 
-    let renderChildren = () => (
-      <Flex direction="column" UNSAFE_className={classNames(labelStyles, 'spectrum-Field-wrapper')}>
-        {children}
-        {hasHelpText && renderHelpText()}
-      </Flex>
-    );
+    let renderChildren = () => {
+      if (labelPosition === 'side') {
```
Comment (snowystinger) [inline packages/@react-spectrum/label/src/Field.tsx:83]: why did this block of code change? label position side has been working so far?

Comment (snowystinger) [reply packages/@react-spectrum/label/src/Field.tsx:83]: if it's related to the test for Field below, I had a bunch of tests I was suggesting to Yihui that may be of interest

https://github.com/adobe/react-spectrum/pull/3320#pullrequestreview-1057973922

obviously you don't need to differentiate with readonly like I was in those

**Review by snowystinger — COMMENTED** [review]
Screen Shot 2022-08-04 at 9 30 47 AM

It appears to be growing at the wrong rate, I typed "hello " and then pasted that text over and over and it starts gaining extra empty lines.

It's happening in Quiet as well.

My guess is that it renders the scrollbar for me, try turning scrollbars on. You should be able to account for them by doing something similar to what I did in Table/ListView https://github.com/adobe/react-spectrum/blob/c0b2454c29360415538a693d4d827b3bce72c303/packages/%40react-spectrum/table/src/TableView.tsx#L291

Once typed into, non-quiet fields never shrink their height, see https://reactspectrum.blob.core.windows.net/reactspectrum/08bef23b0613351ea26d28963432e810edba90d1/storybook/index.html?path=/story/textarea--custom-height-with-label&providerSwitcher-toastPosition=bottom
Type a bunch into the first textarea then delete it all

File: packages/@react-spectrum/label/src/Field.tsx (hunk 1 above)
Comment (devongovett) [reply packages/@react-spectrum/label/src/Field.tsx:83]: This was from Michael's original PR. I think the extra wrapper isn't needed when the label is on top

Comment (devongovett) [reply packages/@react-spectrum/label/src/Field.tsx:83]: Yeah, without this change, the textarea doesn't expand to the full height.

**Review by reidbarber — APPROVED** [review]
LGTM, tested on FF, Safari, Chrome, and iOS Safari.

**Review by snowystinger — APPROVED** [review]
lgtm tested in chrome

my one concern is that people have pre-filled forms without heights set and now the textarea will be very tall

**Pull request merged**